include/huk_load.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef HUK_LOAD_H_
+#define HUK_LOAD_H_
+
+/**
+ * @file
+ * @defgroup huk_load Hardware Unique Key (HUK) loading
+ * @{
+ *
+ * @brief API for loading the hardware unique key (HUK) in the CryptoCell
+ *        KDR registers.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief This function loads the HUK into the KDR registers
+ *        of the Cryptocell. It also locks the flash page which contains
+ *        the key material from readind and writing until next reboot.
+ *
+ * @retval 0        On success.
+ * @retval -EINVAL  If an error occurred.
```

Comment:
In general it is smart to be vague here, to avoid having outdated header files when the implementation changes. Typically these would read something like:

```
 * @return non-negative on success, negative errno code on fail
```

That way it is more future proof in that a value >= 0 will always indicate success, and any negative value is an error.

Currently one could make the argument that the caller only needs to check for -EINVAL.

Reply:
> In general it is smart to be vague here,

Why, that makes the API return code value useless for the programmer.
If I only know that `>= 0` means success, then how should I interpret `5` ?
What would that mean ? Success for sure, but what additional info does 5 carry compared to 0.

Reply:
Then you can interpret it as success, if that is all you need to know, then that is fine.
If you in the future decide that 5 will mean, this passed - and btw some other thing happened, then users can check for 5 explicitly if they need to know if the other thing happened.

You can still list the @retval explicitly, but in its current state it is prone to become out of date if the implementation changes. Currently -EINVAL gives no more info than simply saying that it failed, which is the same level of information you are getting from any negative value.